format/FlightSql.proto

```diff
@@ -1779,9 +1796,60 @@ message CommandPreparedStatementUpdate {
 }
 
 /*
- * Returned from the RPC call DoPut when a CommandStatementUpdate
- * CommandPreparedStatementUpdate was in the request, containing
- * results from the update.
+ * Represents a bulk ingestion request. Used in the command member of FlightDescriptor
+ * for the the RPC call DoPut to cause the server load the contents of the stream's
+ * FlightData into the target destination.
+ */
+message CommandStatementIngest {
+  option (experimental) = true;
+
+  // Describes the behavior for loading bulk data.
+  enum IngestMode {
+    // Ingestion behavior unspecified.
+    INGEST_MODE_UNSPECIFIED = 0;
+    // Create the target table. Fail if the target table already exists.
+    INGEST_MODE_CREATE = 1;
+    // Append to an existing target table. Fail if the target table does not exist.
+    INGEST_MODE_APPEND = 2;
+    // Drop the target table if it exists. Then follow INGEST_MODE_CREATE behavior.
+    INGEST_MODE_REPLACE = 3;
+    // Create the target table if it does not exist. Then follow INGEST_MODE_APPEND behavior.
+    INGEST_MODE_CREATE_APPEND = 4;
+  }
+
+  // The ingestion behavior.
+  IngestMode mode = 1;
+  // The table to load data into.
+  string table = 2;
+  // The db_schema of the destination table to load data into. If unset, a backend-specific default may be used.
+  optional string schema = 3;
+  // The catalog of the destination table to load data into. If unset, a backend-specific default may be used.
+  optional string catalog = 4;
+  /*
+   * Store ingested data in a temporary table.
+   * Leaving temporary unspecified is equivalent to temporary=false.
+   * The effect of setting temporary depends on the choice of IngestMode:
+   *  - INGEST_MODE_CREATE: Create a temp table. Fail if it already exists.
+   *  - INGEST_MODE_APPEND: Add the following validations to the existing INGEST_MODE_APPEND behavior:
+   *    - temporary=true  - Use temporary target table with the given name, error if none exists
+   *    - temporary=false - Use non-temporary target table with the given name, error if none exists
+   *  - INGEST_MODE_REPLACE: Drop the temp table if it exists. Then follow INGEST_MODE_CREATE behavior.
+   *  - INGEST_MODE_CREATE_APPEND: Create a temp table if it does not exist. Then follow INGEST_MODE_APPEND behavior.
+  */
+  optional bool temporary = 5;
```

Comment:
I think a better field name rather then temporary could also help make this field more intuitive to end users.

Reply:
I think (1) this name is 1:1 with SQL temporary tables, and (2) Flight SQL is not meant to be exposed directly to end users

You could possibly argue to put `temporary` in a oneof with `catalog` and `schema`

Reply:
Would perhaps `use_temporary_table` be a better name? And we can remove the docstring and instead clarify that it's equivalent to setting a database-specific catalog/schema pair (i.e. it does not matter if another table exists with the same name in a different catalog/schema).

Reply:
I don't mind `use_temporary_table`. I'll wait for confirmation that this is preferred naming before updating the code. 

> And we can remove the docstring and instead clarify that it's equivalent to setting a database-specific catalog/schema pair

I think this makes the behavior a lot clearer, but what if a particular backend happens to implement temporary tables with a different mechanism? For example, the SQL Server [docs](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16#temporary-tables) indicate that temp tables are implemented "by internally appending a numeric suffix to each local temporary table name". Do you think this should be a behavior we are prescriptive about, or defer to the backend implementation to determine what `temporary` means?

Reply:
I think SQL Server is effectively the same thing, just using a different namespacing mechanism. So you are right in that we can't say that it's 1:1 with setting catalog/schema, but I still would prefer to explain it as a namespace (effectively) instead of trying to write out all the different behaviors.